dot/parachain/candidate-validation/worker_pool.go

```diff
@@ -0,0 +1,163 @@
+package candidatevalidation
+
+import (
+	"fmt"
+
+	parachainruntime "github.com/ChainSafe/gossamer/dot/parachain/runtime"
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+)
+
+type workerPool struct {
+	workers map[parachaintypes.ValidationCodeHash]*worker
+}
+
+type ValidationTask struct {
+	PersistedValidationData parachaintypes.PersistedValidationData
+	WorkerID                *parachaintypes.ValidationCodeHash
+	CandidateReceipt        *parachaintypes.CandidateReceipt
+	PoV                     parachaintypes.PoV
+	ExecutorParams          parachaintypes.ExecutorParams
+	PvfExecTimeoutKind      parachaintypes.PvfExecTimeoutKind
+	ValidationCode          *parachaintypes.ValidationCode
+}
+
+// ValidationResult represents the result coming from the candidate validation subsystem.
+// Validation results can be either a Valid or InvalidValidationResult.
```

Comment:
```suggestion
// Validation results can be either a valid or invalid.
```

There is no need to use exact struct names or function names in comments. In fact, we should avoid that.

Reply:
Ok, updated.